test/integration/shell_int_test.go

```diff
@@ -51,6 +51,31 @@ func (suite *ShellIntegrationTestSuite) TestShell() {
 	}
 }
 
+func (suite *ShellIntegrationTestSuite) TestDefaultShell() {
+	suite.OnlyRunForTags(tagsuite.Shell)
+
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cp := ts.SpawnWithOpts(
+		e2e.WithArgs("activate", "ActiveState-CLI/Python3", "--default", "--path", ts.Dirs.Work),
+		e2e.AppendEnv("ACTIVESTATE_CLI_DISABLE_RUNTIME=false"),
+	)
+	cp.Expect("Activated")
+	cp.WaitForInput()
+	cp.SendLine("exit")
+	cp.ExpectExitCode(0)
+
+	cp = ts.SpawnWithOpts(
+		e2e.WithArgs("shell"),
+		e2e.AppendEnv("ACTIVESTATE_CLI_DISABLE_RUNTIME=false"),
+	)
+	cp.Expect("Activated")
+	cp.WaitForInput()
+	cp.SendLine("exit")
+	cp.ExpectExitCode(0)
+}
```

Comment:
nit: There could be a few more tests that cover the cases in the ACs. This is testing that it sources the global default but we could also have tests for the current project, namespace, current project when there is currently a global default and potentially more.

Reply:
This behavior hinges on `projectfile.GetProjectFilePath()`, which I'm almost positive is tested as a side-effect in other integration tests like `state activate`. I didn't think the additional explicit tests are worth ballooning the integration test runtime even more. What do you think @Naatan ?

Reply:
I agree this test is a little bare. It'd be nice if we can test the "project from CWD" and "project from namespace" variations. 

While technically this is a little redundant with activate, there is enough plumbing here that they aren't identical.

It would also be good if we can test that it actually is in a subshell. The current test just "waits for input" which doesn't assert whether it is now in a subshell.

Reply:
Okay, I added a cwd test. The `cp.Expect("Activated")` is the assertion that we're in a subshell, and the `cp.SendLine("exit")` and subsequent `cp.ExpectExitCode(0)` indirectly verify we've exited the subshell. (If we were not in a subshell, those lines would fail.)

The "project from namespace" variation tests already exist.